client/src/components/ListView/ListView.tsx

```diff
@@ -0,0 +1,67 @@
+import { Avatar, Box, Grid, List, ListItem, ListItemAvatar, ListItemText, Typography } from '@material-ui/core';
+import { Link } from 'react-router-dom';
+import { contest } from '../../interface/tempContestData';
+import React, { memo } from 'react';
+import useStyles from './useStyles';
+
+interface Props {
+  data: contest[];
+  route: string;
+  message: string;
+}
+
+const ListView = memo(function ListView({ data, message, route }: Props): JSX.Element {
+  const classes = useStyles();
+  //Displays all objects within data array and provides additional message in case no objects within array
+  //enables a link to specific details page. <--- Requires Update when integrating Contest/Submissions
+  return (
+    <div>
+      {data.length > 0 ? (
+        <List>
+          {data.map((item) => {
+            return (
+              <Link className={classes.link} to={route} key={item.id}>
+                <ListItem key={item.id}>
+                  <ListItemAvatar>
+                    <Avatar className={classes.img} alt="Profile Image" src={item.images} variant="square" />
+                  </ListItemAvatar>
+                  <Grid
+                    className={classes.ListItems}
+                    container
+                    direction="column"
+                    justify="flex-start"
+                    alignItems="flex-start"
+                  >
+                    <ListItemText
+                      primary={
+                        <React.Fragment>
```

Comment:
do we need this extra Fragment?

Reply:
absolutely not I will remove it right away !